frontend/packages/console-shared/src/utils/resource-utils.ts

```diff
@@ -159,23 +161,22 @@ const sortByRevision = (
   return _.toArray(replicators).sort(compare);
 };
 
-const getAnnotation = (obj: K8sResourceKind, annotation: string): string => {
-  return _.get(obj, ['metadata', 'annotations', annotation]);
+const getAnnotation = (obj: K8sResourceCommon, annotation: string): string => {
+  return obj?.metadata?.annotations?.[annotation];
 };
 
-const getDeploymentRevision = (obj: K8sResourceKind): number => {
+export const getDeploymentRevision = (obj: K8sResourceCommon): number => {
   const revision = getAnnotation(obj, DEPLOYMENT_REVISION_ANNOTATION);
   return revision && parseInt(revision, 10);
 };
 
-const getDeploymentConfigVersion = (obj: K8sResourceKind): number => {
+export const getDeploymentConfigVersion = (obj: K8sResourceCommon): number => {
   const version = getAnnotation(obj, DEPLOYMENT_CONFIG_LATEST_VERSION_ANNOTATION);
   return version && parseInt(version, 10);
 };
 
-const getDeploymentConfigName = (obj: K8sResourceKind): string => {
-  return _.get(obj, 'metadata.ownerReferences[0].name', null);
-};
+export const getOwnerNameByKind = (obj: K8sResourceCommon, kind: string): string =>
+  obj?.metadata?.ownerReferences?.find((ref) => ref.kind === kind)?.name;
```

Comment:
If we're making this a common utility, we should really require the API group, not just the kind. The kind on its own doesn't identify a resource.